pgrx/src/datum/array.rs

```diff
@@ -495,9 +487,19 @@ mod casper {
     }
 
     pub(super) struct PassByFixed {
-        pub(super) align: usize,
-        pub(super) size: usize,
+        pub(super) padded_size: usize,
+    }
+
+    impl PassByFixed {
+        pub(super) fn from_size_and_align(size: u16, align: Align) -> Self {
```

Comment:
Chatted on discord about this impl, had a few tweaks. Also please `#[inline]` this function so iterators in external crates can see how the size was computed (admittedly after devirtualizing).